PWGMM/UE/Tasks/lambdak0sflattenicity.cxx

```diff
@@ -0,0 +1,826 @@
+          i_ring = 20;
+        } else if (i_ch == 22) {
+          i_ring = 21;
+        } else if (i_ch == 21) {
+          i_ring = 22;
+        } else if (i_ch == 20) {
+          i_ring = 23;
+        }
+      }
+    } else if (i_ch >= 24 && i_ch < 32) {
+      if (i_ch < 28) {
+        i_ring = i_ch;
+      } else {
+        if (i_ch == 31) {
+          i_ring = 28;
+        } else if (i_ch == 30) {
+          i_ring = 29;
+        } else if (i_ch == 29) {
+          i_ring = 30;
+        } else if (i_ch == 28) {
+          i_ring = 31;
+        }
+      }
+    } else if (i_ch == 32) {
+      i_ring = 32;
+    } else if (i_ch == 40) {
+      i_ring = 33;
+    } else if (i_ch == 33) {
+      i_ring = 34;
+    } else if (i_ch == 41) {
+      i_ring = 35;
+    } else if (i_ch == 34) {
+      i_ring = 36;
+    } else if (i_ch == 42) {
+      i_ring = 37;
+    } else if (i_ch == 35) {
+      i_ring = 38;
+    } else if (i_ch == 43) {
+      i_ring = 39;
+    } else if (i_ch == 47) {
+      i_ring = 40;
+    } else if (i_ch == 39) {
+      i_ring = 41;
+    } else if (i_ch == 46) {
+      i_ring = 42;
+    } else if (i_ch == 38) {
+      i_ring = 43;
+    } else if (i_ch == 45) {
+      i_ring = 44;
+    } else if (i_ch == 37) {
+      i_ring = 45;
+    } else if (i_ch == 44) {
+      i_ring = 46;
+    } else if (i_ch == 36) {
+      i_ring = 47;
+    }
+    return i_ring;
+  }
+
+  float GetFlatenicity(float signals[], int entries)
+  {
+    float flat = 9999;
+    float mRho = 0;
+    for (int iCell = 0; iCell < entries; ++iCell) {
+      mRho += 1.0 * signals[iCell];
+    }
+    // average activity per cell
+    mRho /= (1.0 * entries);
+    // get sigma
+    float sRho_tmp = 0;
+    for (int iCell = 0; iCell < entries; ++iCell) {
+      sRho_tmp += TMath::Power(1.0 * signals[iCell] - mRho, 2);
+    }
+    sRho_tmp /= (1.0 * entries * entries);
+    float sRho = TMath::Sqrt(sRho_tmp);
+    if (mRho > 0) {
+      flat = sRho / mRho;
+    }
+    return flat;
+  }
+
+  // Defining filters for events (event selection)
+  // Processed events will be already fulfilling the event selection requirements
+  Filter eventFilter = (o2::aod::evsel::sel8 == true);
+  Filter posZFilter = (nabs(o2::aod::collision::posZ) < cutzvertex);
+
+  // Filters on V0s
+  // Cannot filter on dynamic columns, so we cut on DCA to PV and DCA between daughters only
+  Filter preFilterV0 = (nabs(aod::v0data::dcapostopv) > v0setting_dcapostopv &&
+                        nabs(aod::v0data::dcanegtopv) > v0setting_dcanegtopv &&
+                        aod::v0data::dcaV0daughters < v0setting_dcav0dau);
+
+  // Defining the type of the daughter tracks
+  using DaughterTracks = soa::Join<aod::TracksIU, aod::TracksExtra, aod::pidTPCPi, aod::pidTPCPr>;
+
+  Filter trackFilter = (nabs(aod::track::eta) < cfgTrkEtaCut);
+  using TrackCandidates = soa::Filtered<soa::Join<aod::TracksIU, aod::Tracks, aod::TracksExtra, aod::TracksDCA, aod::TrackSelection, aod::pidTPCPi, aod::pidTPCPr>>;
```

Comment:
Joining `TracksIU` and `Tracks` is incorrect, since they have essentially identical set of columns, only one is actually used. These are not supposed to be joined, you should use one or the other.

Reply:
I have kept only TracksUI now. Thank you for the correction.